kendzi3d-plugin/src/main/java/kendzi/josm/kendzi3d/jogl/model/ground/GroundDrawer.java

```diff
@@ -60,7 +61,7 @@ public void draw(GL2 gl, Point3d cameraPosition) {
         double zButtom = c.z + groundSize;
         double zTop = c.z - groundSize;
 
-        gl.glNormal3d(0, 1, 0);
+        gl.glNormal3d(0, modelRender.isDoubleSided()?-1:1, 0);

```

Comment:
why you check double sided for drawing ground?

Reply:
isDoubleSided() is true if  the user enabled  double sided lighting with the 3D menu in JOSM.

Double sided lighting is more expensive than doing only single side light calculation for a face.  If only a single side was light-calculated, but both sides of the face may be viewed, they are both lit in the same way.

IF face culling works, you actually do not need it, because in this case the only side ever shown of a face is the front side, so by default this feature is turned off.  If the back side is never shown, doing a light calculation for that side is not necessary.

I experimented with not using face culling at all.  A single face to represent both sides of a fence/wall may be used.  In this case the front side and the back side of the face are shown, depending on the view.  And in this case double sided lighting makes sense to turn on (to have the back side of the face lit differently than the front side).

----
It actually makes more sense to add a field  'useDoubleSided' to Model.java, and enable double sided lighting on a  ''per model basis'', instead of globally for all models.   But during experimentation I enabled it either for all models or none at all.

If double sided lighting is used with the ground plane, the normal vector has to be set up for the front side (which points in negative Y direction), for the back side to correctly lit  (if double sided lighting is turned on, OpenGL will flip the normal of the front side  automatically to determine back side lighting.).   An alternative might have been to use a different winding when drawing the ground plane so that GL recognizes the front side pointing in positive Y.

If we enable double sided lighting on a per model basis  like suggested above, then this change will not be needed (that is if it will then never be turned on for the ground model).

----
I still think that it would be more performant to draw fences with only a single face for both sides and turn on double sided lighting (just for these models), than to use face culling.  But for now I made it work with face culling, to maintain the logic you originally set up to draw this  (and that did not draw correctly on my setup without patching..).

Reply:
The comment may repeat lots of stuff you already know, but I think its better to fully light the details when reviewing code and changes made than to discuss on differing (but unknown) assumptions made.

So sorry for the length of the comment nonetheless, it might have been to elaborate.